pkg/reconciler/backup/volumesnapshot/reconciler.go

```diff
@@ -505,29 +513,89 @@ func (se *Reconciler) waitSnapshotToBeProvisionedAndAnnotate(
 ) (*ctrl.Result, error) {
 	contextLogger := log.FromContext(ctx)
 
+	// Extract the retry configuration from the cluster
+	var retryConfig *apiv1.VolumeSnapshotRetryConfiguration
+	if clusterName, ok := snapshot.Labels[utils.ClusterLabelName]; ok {
+		var cluster apiv1.Cluster
+		if err := se.cli.Get(ctx, client.ObjectKey{
+			Namespace: snapshot.Namespace,
+			Name:      clusterName,
+		}, &cluster); err == nil {
+			if cluster.Spec.Backup != nil &&
+				cluster.Spec.Backup.VolumeSnapshot != nil &&
+				cluster.Spec.Backup.VolumeSnapshot.RetryConfiguration != nil {
+				retryConfig = cluster.Spec.Backup.VolumeSnapshot.RetryConfiguration
+			}
+		}
+	}
+
 	info := parseVolumeSnapshotInfo(snapshot)
 	if info.error != nil {
-		if info.error.isRetryable() {
+		if info.error.isRetryable(retryConfig, info.retryCount) {
+			// Record metrics for this retry
+			metrics.RecordRetry(snapshot.Namespace, snapshot.Name, metrics.PhaseProvisioning)
+
+			// Record error-specific metrics
+			errorType := "unknown"
+			if err := info.error; err != nil {
+				if errors.Is(err, context.DeadlineExceeded) {
+					errorType = "context.DeadlineExceeded"
+				} else if errors.Is(err, context.Canceled) {
+					errorType = "context.Canceled"
+				} else {
+					errorType = fmt.Sprintf("%T", err)
+				}
+			}
+			metrics.RecordRetryWithError(snapshot.Namespace, snapshot.Name,
+				metrics.PhaseProvisioning, errorType)
+
```

Comment:
I think we should record any relevant data in the status without invoking any metrics pkg. 

Once the metrics status spins up it calculates the metrics from the available cluster status.

This avoids potentially mixing up an observability layer with other layers